Assets/__Scripts/Platforms/PlatformDescriptor.cs

```diff
@@ -260,6 +260,21 @@ void HandleLights(LightsManager group, int value, MapEvent e)
         if (SoloAnEventType && e._type != SoloEventType) mainColor = invertedColor = Color.black.WithAlpha(0);
 
         IEnumerable<LightingEvent> allLights = group.ControllingLights;
+
+        if ((e._customData?.HasKey("_lightID") ?? false) && Settings.Instance.EmulateChromaAdvanced)
+        {
+            int lightID = group.EditorToGameLightIDMap.IndexOf(e._customData["_lightID"].AsInt);
+            if (lightID >= 0 && lightID < group.ControllingLights.Count)
+            {
+                allLights = new List<LightingEvent> {group.ControllingLights[lightID]};
```

Comment:
Is constructing a new list with every lightID event necessary? From experience doing something similar to this in previous ChroMapper versions, `List<T>`s can be pretty expensive to construct. We should be able to just assign straight into the `allLights` variable because it is a `IEnumerable<LightingEvent>`.